zulip/zulip/__init__.py

```diff
@@ -326,6 +328,25 @@ def validate_boolean_field(field: Optional[str]) -> Union[bool, None]:
         return None
 
 
+FuncT = Callable[..., object]
+
+
+def compatible(
```

Comment:
It'd be great to add a comment/docstring here.  I also think it might be appropriate to make the `min_feature_level` / `max_feature_level` parameters mandatory kwargs (i.e. `alternative: FuncT, *, max_feature_level: int=...`.

Reply:
Yes, I think we will turn to the `if` approach after the discussion.